tools/structures.go

```diff
@@ -0,0 +1,597 @@
+package tools
+
+import (
+	"bufio"
+	"errors"
+	"math"
+	"strconv"
+	"strings"
+)
+
+var conduitShapes map[int]string = map[int]string{
```

Comment:
It looks like you are passing htis around as either shapeMap or passing it directly to functions as conduitShapes.  If you are making this map global, you don't need to pass it through any functions, any code within the tools package should be able to access it directly.

Reply:
Good point. I no longer pass this map and the unitConsistencyMap map around but call them as global variables.